pallets/chainbridge/src/tests.rs

```diff
@@ -0,0 +1,154 @@
+	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
+	type PalletInfo = PalletInfo;
+	type BlockWeights = ();
+	type BlockLength = ();
+	type SS58Prefix = ();
+	type OnSetCode = ();
+	type MaxConsumers = ConstU32<2>;
+}
+
+parameter_types! {
+	pub const ExistentialDeposit: u64 = 1;
+}
+
+ord_parameter_types! {
+	pub const One: u64 = 1;
+}
+
+impl pallet_balances::Config for Test {
+	type Balance = u64;
+	type DustRemoval = ();
+	type Event = Event;
+	type ExistentialDeposit = ExistentialDeposit;
+	type AccountStore = System;
+	type WeightInfo = ();
+	type MaxLocks = ();
+	type MaxReserves = ();
+	type ReserveIdentifier = [u8; 8];
+}
+
+parameter_types! {
+	pub const TestChainId: u8 = 5;
+	pub const ProposalLifetime: u64 = 50;
+}
+
+impl Config for Test {
+	type Event = Event;
+	type BridgeCommitteeOrigin = frame_system::EnsureRoot<Self::AccountId>;
+	type Proposal = Call;
+	type BridgeChainId = TestChainId;
+	type ProposalLifetime = ProposalLifetime;
+}
+
+// pub const BRIDGE_ID: u64 =
+pub const RELAYER_A: u64 = 0x2;
+pub const RELAYER_B: u64 = 0x3;
+pub const RELAYER_C: u64 = 0x4;
+pub const ENDOWED_BALANCE: u64 = 100_000_000;
+pub const TEST_THRESHOLD: u32 = 2;
+
+pub fn new_test_ext() -> sp_io::TestExternalities {
+	let bridge_id = PalletId(*b"phala/bg").into_account();
+	let mut t = frame_system::GenesisConfig::default().build_storage::<Test>().unwrap();
+	pallet_balances::GenesisConfig::<Test> { balances: vec![(bridge_id, ENDOWED_BALANCE)] }
+		.assimilate_storage(&mut t)
+		.unwrap();
+	let mut ext = sp_io::TestExternalities::new(t);
+	ext.execute_with(|| System::set_block_number(1));
+	ext
+}
+
+pub fn new_test_ext_initialized(
+	src_id: BridgeChainId,
+	r_id: ResourceId,
+	resource: Vec<u8>,
+) -> sp_io::TestExternalities {
+	let mut t = new_test_ext();
+	t.execute_with(|| {
+		// Set and check threshold
+		assert_ok!(Bridge::set_threshold(Origin::root(), TEST_THRESHOLD));
+		assert_eq!(Bridge::relayer_threshold(), TEST_THRESHOLD);
+		// Add relayers
+		assert_ok!(Bridge::add_relayer(Origin::root(), RELAYER_A));
+		assert_ok!(Bridge::add_relayer(Origin::root(), RELAYER_B));
+		assert_ok!(Bridge::add_relayer(Origin::root(), RELAYER_C));
+		// Whitelist chain
+		assert_ok!(Bridge::whitelist_chain(Origin::root(), src_id));
+		// Set and check resource ID mapped to some junk data
+		assert_ok!(Bridge::set_resource(Origin::root(), r_id, resource));
+		assert_eq!(Bridge::resource_exists(r_id), true);
+	});
+	t
+}
+
+// Checks events against the latest. A contiguous set of events must be provided. They must
+// include the most recent event, but do not have to include every past event.
+pub fn assert_events(mut expected: Vec<Event>) {
+	let mut actual: Vec<Event> =
+		system::Pallet::<Test>::events().iter().map(|e| e.event.clone()).collect();
+
+	expected.reverse();
+
+	for evt in expected {
+		let next = actual.pop().expect("event expected");
+		assert_eq!(next, evt.into(), "Events don't match (actual,expected)");
+	}
+}
```

Comment:
Content of this file is the same as `pallets/chainbridge/src/mock.rs` I assume that this is an adaptation of [ChainSafe/chainbridge-substrate](https://github.com/ChainSafe/chainbridge-substrate) repo and this tests module should contain adapted tests from  [here](https://github.com/ChainSafe/chainbridge-substrate/blob/main/chainbridge/src/tests.rs).